SteamBot/UserHandler.cs

```diff
@@ -193,14 +193,20 @@ public virtual void OnTradeRequestReply(bool accepted, string response)
         #region Trade events
         // see the various events in SteamTrade.Trade for descriptions of these handlers.
 
-        public abstract void OnTradeError (string error);
-
-        public virtual void OnStatusError(Trade.TradeStatusType status)
-        {
-            string otherUserName = Bot.SteamFriends.GetFriendPersonaName(OtherSID);
-            string statusMessage = (Trade != null ? Trade.GetTradeStatusErrorString(status) : "died a horrible death");
-            string errorMessage = String.Format("Trade with {0} ({1}) {2}", otherUserName, OtherSID.ConvertToUInt64(), statusMessage);
-            OnTradeError(errorMessage);
+        public abstract void OnTradeError (SteamBotError error);
+
+        public virtual void OnStatusError (Trade.TradeStatusType status)
+		{
```

Comment:
Your indentation is messed up here.  Make sure to set VS to change tabs to spaces.